apps/examples/camera/camera_main.c

```diff
@@ -0,0 +1,382 @@
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static int write_file(uint8_t *data, size_t len, uint32_t format);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static struct v_buffer *buffers_video;
+static struct v_buffer *buffers_still;
+static unsigned int n_buffers;
+
+static uint8_t camera_main_file_count = 0;
+static char camera_main_filename[IMAGE_FILENAME_LEN];
+static const char *save_dir;
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+static int write_file(uint8_t *data, size_t len, uint32_t format)
+{
+	FILE *fp;
+
+	camera_main_file_count++;
+	if (camera_main_file_count >= 1000) {
+		camera_main_file_count = 1;
+	}
+
+	memset(camera_main_filename, 0, sizeof(camera_main_filename));
+
+	if (format == V4L2_PIX_FMT_JPEG) {
+		snprintf(camera_main_filename, IMAGE_FILENAME_LEN, "%s/VIDEO%03d.JPG", save_dir, camera_main_file_count);
+	} else {
+		snprintf(camera_main_filename, IMAGE_FILENAME_LEN, "%s/VIDEO%03d.YUV", save_dir, camera_main_file_count);
+	}
+
+	fprintf(stderr, "FILENAME:%s\n", camera_main_filename);
+
+	fp = fopen(camera_main_filename, "wb");
+	if (NULL == fp) {
+		fprintf(stderr, "fopen error : %d\n", errno);
+		return -1;
+	}
+
+	if (len != fwrite(data, 1, len, fp)) {
+		fprintf(stderr, "fwrite error : %d\n", errno);
+	}
+
+	fclose(fp);
+	return 0;
+}
+
+static int camera_prepare(int fd, enum v4l2_buf_type type, uint32_t buf_mode, uint32_t pixformat, uint16_t hsize, uint16_t vsize, uint8_t buffernum)
+{
+	int ret;
+	int cnt;
+	uint32_t fsize;
+	struct v4l2_format fmt = { 0 };
+	struct v4l2_requestbuffers req = { 0 };
+	struct v4l2_buffer buf = { 0 };
+	struct v_buffer *buffers;
+
+	/* VIDIOC_REQBUFS initiate user pointer I/O */
+
+	req.type = type;
+	req.memory = V4L2_MEMORY_USERPTR;
+	req.count = buffernum;
+	req.mode = buf_mode;
+
+	ret = ioctl(fd, VIDIOC_REQBUFS, (unsigned long)&req);
+	if (ret < 0) {
+		fprintf(stderr, "Failed to VIDIOC_REQBUFS: errno = %d\n", errno);
+		return ret;
+	}
+
+	/* VIDIOC_S_FMT set format */
+
+	fmt.type = type;
+	fmt.fmt.pix.width = hsize;
+	fmt.fmt.pix.height = vsize;
+	fmt.fmt.pix.field = V4L2_FIELD_ANY;
+	fmt.fmt.pix.pixelformat = pixformat;
+
+	ret = ioctl(fd, VIDIOC_S_FMT, (unsigned long)&fmt);
+	if (ret < 0) {
+		fprintf(stderr, "Failed to VIDIOC_S_FMT: errno = %d\n", errno);
+		return ret;
+	}
+
+	/* VIDIOC_QBUF enqueue buffer */
+
+	if (type == V4L2_BUF_TYPE_VIDEO_CAPTURE) {
+		buffers_video = malloc(sizeof(v_buffer_t) * buffernum);
```

Comment:
You should check if memory allocation succeeded.

Reply:
Check at line number 166